ndcube/mixins/ndcubeio.py

```diff
@@ -0,0 +1,257 @@
+        if self.__meta is not None:
+            header = fits.Header(self.__meta.copy())
+
+        else:
+            header = fits.Header()
+        # Create a FITS Safe header from the meta data
+
+        # Cross-check if the metadata is fits safe, if not then it is corrected
+        for k, v in header.items():
+            _insert_in_metadata_fits_safe(header, k, v)
+
+        if self.__wcs:
+            # Create a header for a given wcs object
+            # Hard-Coded relax parameter to write all
+            # recognized informal extensions of the WCS standard.
+            wcs_header = self.__wcs.to_header(relax=True)
+            header.extend(wcs_header, useblanks=False, update=True)
+
+        # Create a FITS header for storing missing_axis
+        # MISSING0 : 1 if axis 1 is missing
+        # other keywords are skipped, if the axis is present
+        if self.__missing_axis:
+            header0 = fits.Header()
+            for index, axis in enumerate(self.__missing_axis):
+                if axis:
+                    header0[MISSING.format(index)] = 1
+                else:
+                    continue
+
+            for k, v in header0.items():
+                _insert_in_metadata_fits_safe(header0, k, v)
+
+            header.extend(header0, useblanks=False, update=True)
+
+
+        # PrimaryHDU list contains only meta, missing_axis and wcs as headers, no data
+        hdus = [fits.PrimaryHDU(data=None, header=header)]
+
+        #------------------------------HDU0----------------------------------
+
+        #------------------------------HDU1----------------------------------
+        # Header for unit
+        if self.__unit:
+            header_unit = fits.Header()
+            if self.__unit is not u.dimensionless_unscaled:
+                header_unit['bunit'] = self.__unit.to_string()
+            hdus.append(fits.ImageHDU(self.__data, header_unit, name='UNIT'))
+
+        #------------------------------HDU1----------------------------------
+
+        #------------------------------HDU2----------------------------------
+        # Store the uncertainty
+        if self.__uncertainty is not None:
+            
+            # Set the initial header, and the type of uncertainty
+            hdr_uncertainty = fits.Header()
+            hdr_uncertainty['UTYPE'] = self.__uncertainty.uncertainty_type
+
+            # Set the data of the uncertainty and header
+            hduUncert = fits.ImageHDU(self.__uncertainty.array, hdr_uncertainty, name='UTYPE')
+            hdus.append(hduUncert)
+        #----------------------------HDU2--------------------------------
+
+        #----------------------------HDU3------------------------------
+        # Store the mask
+        if self.__mask is not None:
+            # Always assuming that the mask is a np.ndarray
+            # by checking that it has a shape
+            if not hasattr(self.__mask, 'shape'):
```

Comment:
Instead of looking for a attribute and assuming ```self.__mask``` is the right type, use ```isinstance to actually check the type.
```python
if not isinstance(self.__mask, np.ndarray):
```

Reply:
That would be elegant.